csrc/optimization/layout_optimization.cpp

```diff
@@ -0,0 +1,45 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <optimization/layout_inference.h>
+#include <optimization/layout_optimization.h>
+
+namespace nvfuser::optimization {
+
+void LayoutOptimizationPass::runPass(Fusion* fusion) {
+  std::unordered_map<const TensorView*, MemoryFormat> stride_mapping =
+      inferenceMemoryFormat(fusion);
+
+  for (auto out_val : fusion->outputs()) {
+    auto out_tv = dynamic_cast<TensorView*>(out_val);
+    // skip:
+    //   1. non-tensor output;
+    //   2. tensor output with allocation specified, assuming everything is
+    //   semantical
+    //   3. tensor output that's aliasing (Does aliased src matter?)
```

Comment:
> Does aliased src matter?

No. That's mainly for error checking -- in allocateOutput, we check the aliasing output is indeed an alias of the aliased input.

Reply:
Should I not rely on `getOutputAlias` to see if an output is just an alias? I'm just trying to avoid changing allocation domain on tensor that could have been just an alias

Reply:
Sorry, my earlier comment was confusing. 

There's nothing wrong with this code as is, and you don't need to check `aliased_io`.